autoprecompiles/src/constraint_optimizer.rs

```diff
@@ -335,6 +338,7 @@ fn remove_redundant_constraints<P: FieldElement, V: Clone + Ord + Hash + Display
         .enumerate()
         .map(|(i, c)| {
             let factors = c.to_factors();
+            assert!(!factors.is_empty());
```

Comment:
Is this an invariant of `to_factors`, or of this particular call?

Reply:
I think it would only empty on constraints of the form `<const> = 0`. Seems like we rely on them being filtered out.